packages/metrics/src/config/routes.ts

```diff
@@ -82,13 +83,30 @@ export enum EventType {
   MARRIAGE = 'marriage'
 }
 
+/*
+ * This route wrapper triggers a view update for materialised views
+ * we read data to Metabase from. If you are not seeing your data updated
+ * after a user action, you most likely need to add this wrapper to some
+ * new endpoint handler here.
+ */
+function analyticsDataRefreshingRoute<T extends Array<any>, U>(
+  handler: (...args: T) => U
+) {
+  // Do not use await for the refresh operation. This operation can take minutes or more.
+  // Consider triggering this a task that will be left to be run in the background.
+  return (...params: T) => {
+    refresh()
+    return handler(...params)
+  }
+}
```

Comment:
I wonder how this would affect the data generator if each event that comes to metrics triggers this refresh (also applicable if there is high traffic i.e. lots of declarations are being created). Bunch of refreshes would get queued up. Maybe we should add a `throttle` here?